### PR TITLE
Add column and drop column from an object list or 2D array using ArrayHelper

### PR DESCRIPTION
**Pull request by izharaazmi**
**IMO, this is not ready to be merged yet. No test case written**

I wrote this long time ago, however didn't ever get time to write the test cases. Therefore I am pushing it here if someone would like to carry over the work. If I get a leisure any sooner I'd look into that.

### Summary of Changes
* Add features to add a new column to a list of objects or a 2-D array.
* Add features to drop an entire column from a list of objects or a 2-D array.

### Testing Instructions
Load a list of records from any table (say 10 records from `#__extensions`).

Calling following should remove the `manifest_cache` column from all records in the array:
```php
ArrayHelper::dropColumn($items, 'manifest_cache');
```
Similarly, calling this will add a new column `test` to the list items where the value will be taken from the `$column` array. If `$column`'s key match the `$items`'s `extension_id` value for an item then the `$column`'s respective value will be put as `test` columns value.

In simple terms, `$column` is `array(reference_key_value => new_column_value);`.

If last parameter (optional) is omitted then the `$column` will be looked up using the array index of each item in the source array.

```php
ArrayHelper::addColumn($items, $column, 'test', 'extension_id');
```

### Documentation Changes Required
Add documentation for these two methods.

### CODE REVIEW

**Comment by mbabker** [comment]
Can you add test cases and I'll merge this?

**Comment by izharaazmi** [comment]
@mbabker I have written the test cases. Can you please review?
